src/include/btree_inline.h

```diff
@@ -2118,27 +2118,6 @@ __wt_page_swap_func(WT_SESSION_IMPL *session, WT_REF *held, WT_REF *want, uint32
     return (ret);
 }
 
-/*
- * __wt_btcur_bounds_early_exit --
- *     Performs bound comparison to check if the key is within bounds, if not, increment the
- *     appropriate stat, early exit, and return WT_NOTFOUND.
- */
-static inline int
```

Comment:
Could you explain why you chose to remove this helper function?

I'd have been inclined to try to go the other way. Make this a generic bounds checking function, rather than unwind it into callers.

It could become something like:

```
__wt_btcur_bounds_early_exit(...)
{
    if (!WT_CURSOR_BOUNDS_SET(cbt))
        return (0);

    if ((next && !F_ISSET((cursor), WT_CURSTD_BOUND_UPPER) ||
        (!next && !F_ISSET((cursor), WT_CURSTD_BOUND_LOWER))
         return (0);

    /* As the function used to be */
}
```

Then next, prev and search can all just do:
```
WT_RET(__wt_btcur_bounds_early_exit(session, cbt, false, key_out_of_boundsp));
```

Without any additional conditional checking in those functions. I also don't like seeing "row" specific calls spread around - so having a single function where we'll need to add in column store support is appealing to me.

Reply:
Our initial idea with removing the helper function was to avoid having functions that did little/nothing, and as part of this PR, there was another function planned that would’ve essentially been calling `__wt_btcur_bounds_early_exit` and incrementing a different stat. Due to the stats, combining the functions was a little messy so we decided to move out the stat incrementation to wherever it was happening instead. This is a good point though, and I agree with the value of making this a generic bounds function. I will implement this change!